packages/build/src/plugins/child/error.js

```diff
@@ -13,11 +14,31 @@ const handleError = async function({
   [ERROR_TYPE_SYM]: type = DEFAULT_ERROR_TYPE,
   ...errorProps
 }) {
-  await sendEventToParent('error', { name, message, stack, type, errorProps })
+  const errorPropsA = serializeErrorProps(errorProps)
+  await sendEventToParent('error', { name, message, stack, type, errorProps: errorPropsA })
 }
 
 const DEFAULT_ERROR_TYPE = 'pluginInternal'
 
+// IPC uses JSON for the payload. We ensure there are not circular references
+// as those would make the message sending fail.
+const serializeErrorProps = function(errorProps) {
+  try {
+    return JSON.parse(JSON.stringify(errorProps))
```

Comment:
Could we use something like https://github.com/bugsnag/safe-json-stringify to keep nested properties?
Not sure this is a valid use case.
For example:
```js
const  errorProps  = { a: { b: {} } }
errorProps.a.b.c = errorProps
```

We could potentially keep everything except `c`

Reply:
You're right, let's not reinvent the wheel. Fixed!